provider_shopper/lib/screens/catalog.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+
+class MyCatalog extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: CustomScrollView(
+        slivers: [
+          _MyAppBar(),
+          SliverToBoxAdapter(child: SizedBox(height: 12)),
+          SliverList(
+              delegate:
+                  SliverChildBuilderDelegate((_, index) => _MyListItem(index))),
+        ],
+      ),
+    );
+  }
+}
+
+class _AddButton extends StatelessWidget {
+  final Item item;
+
+  const _AddButton({Key key, @required this.item}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var cart = Provider.of<CartModel>(context);
+
+    return FlatButton(
+      onPressed: cart.items.contains(item) ? null : () => cart.add(item),
+      splashColor: Theme.of(context).primaryColor,
+      child: cart.items.contains(item)
+          ? Icon(Icons.check, semanticLabel: 'ADDED')
+          : Text('ADD'),
+    );
+  }
+}
+
+class _MyAppBar extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return SliverAppBar(
+      title: Text('Catalog', style: Theme.of(context).textTheme.display4),
+      floating: true,
+      actions: [
+        IconButton(
+            icon: Icon(Icons.shopping_cart),
+            onPressed: () => Navigator.pushNamed(context, '/cart')),
```

Comment:
nit: throw a comma after the close of pushNamed so this will indent a little better.